Code/Framework/AzCore/Tests/Settings/ConfigurableStackTests.cpp

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/JSON/document.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/ConfigurableStack.h>
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <AzCore/Serialization/Json/RegistrationContext.h>
+#include <AzCore/Serialization/Json/JsonSystemComponent.h>
+#include <AzCore/UnitTest/TestTypes.h>
+
+namespace UnitTest
+{
+    struct ConfigInt
+    {
+        AZ_TYPE_INFO(UnitTest::ConfigInt, "{1FAF6E55-7FA4-4FFA-8C41-34F422B8E8AB}");
+        AZ_CLASS_ALLOCATOR(ConfigInt, AZ::SystemAllocator, 0);
+
+        int m_value;
+
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto sc = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                sc->Class<ConfigInt>()->Field("Value", &ConfigInt::m_value);
+            }
+        }
+    };
+
+    struct ConfigurableStackTests : public AllocatorsFixture
```

Comment:
FYI the `ScopedAllocatorFixture` is the preferred way to initialize the AZ Allocators.

It doesn't require override the Setup/TearDown function to allocate memory after the base class `AllocatorsFixture::SetUp` has been called and to deallocate memory before the `AllocatorsFixture::TearDown` has been called.

It also allows for value types which uses the AZ System Allocator to be members of the fixture.

Reply:
I wasn't aware of this either, I do know using `AllocatorsTestFixture` is preferred (I think the old name was just left for compatibility with old code) but `ScopedAllocatorFixture` sounds even better. Maybe this should be added to the testing docs if it's not there already?

Reply:
There are testing docs?
If so they should be added there.
Also can you link me those documents?

Reply:
There are! Though I don't think the merge has happened yet. I'd keep an eye on https://github.com/o3de/o3de.org/pull/1394 and maybe see if the recommendation could fit somewhere in there potentially? Might be worth reaching out to @Kadino about it next week.

Reply:
There's already a place for AZTest/GoogleTest docs, which would be the appropriate place to add allocator info: https://www.o3de.org/docs/user-guide/testing/aztest/aztest/

The linked PR is for a getting started guide, which isn't the right place for this detail.